test/shared/src/main/scala/zio/test/laws/ZLawsF.scala

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright 2019-2020 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.test.laws
+
+import zio.ZIO
+import zio.random.Random
+import zio.test.{ check, checkM, Gen, TestResult }
+
+object ZLawsF {
+
+  sealed trait Covariant[-Caps[_[+_]], -R] { self =>
```

Comment:
Maybe we end up with something like:

```suggestion
  sealed trait Covariant[-CapsF[_[+_]], -Caps[_], -R] { self =>
```

So you can do things like `Covariant[Foldable, Equal, Any]`, which would say, my `Foldable` laws can operate on any `A` for which `Equal` is defined. 

Or maybe there's a simpler way.